### ISSUE TITLE
CMake Error

### ISSUE THREAD

**Issue by MuazzezA** [margelo/react-native-worklets-core#189]
**Issue**

<img width="1078" alt="Screenshot " src="https://github.com/luicfrr/react-native-vision-camera-face-detector/assets/64336826/f6d03d64-7828-4d8c-add4-6a5d8b2f9091">

I did all the operations like clearing the cache.

**Device:**
 - Device: emulator
 - OS: android

**`package.json` file:**

```
{
   "dependencies": {
    "react-native-vision-camera": "^3.9.2",
    "react-native-vision-camera-face-detector": "^1.6.3",
    "react-native-worklets-core": "^1.0.0",
  }
}

```

**babel.congif.js**

```
module.exports = {
  presets: ['module:@react-native/babel-preset'],
  plugins: [
    ['react-native-worklets-core/plugin'],
    ['react-native-reanimated/plugin'],
  ],
};

```

**Comment by LongNguyen2312** [comment]
I also have the same problem, does anyone have a solution?

**Comment by eisodev** [comment]
Had similar issues with worklets package:
```
Error: Invalid value for "PACKAGE_PATH": Directory "/Users/xxxx/xxxx/xxx/node_modules/react-native-worklets-core/android/build/intermediates/prefab_package/debug/prefab" is not readable.
```

I updated to the following versions released recently, and my problem was fixed:
```
"react-native-vision-camera": "^4.0.5"
"react-native-worklets-core": "1.3.3"
```

**Comment by sandeep36butte** [comment]
Not sure whethere is's an issue with gradle version, filetree filter method seems not working
There's an prepare header task in the gradle file which copy all the cpp files to the rnworkletcore header dir in build which is not success.

Here's a patch which will copy the files individual, this fixed issue for me.
FileLocation: "../node_modules/react-native-worklets-core/android/build.gradle"

` task prepareHeaders(type: Copy) {
-  from fileTree('../cpp').filter { it.isFile() }
-  into "${project.buildDir}/headers/rnworklets/react-native-worklets-core/"
+  fileTree('../cpp').each{file ->
+    println "file: ${file.isFile()}"
+    copy {
+      from file
+      into "${project.buildDir}/headers/rnworklets/react-native-worklets-core/"
+    }
+  }
   includeEmptyDirs = false
 }`

**Comment by devtyty** [comment]
> Not sure whethere is's an issue with gradle version, filetree filter method seems not working There's an prepare header task in the gradle file which copy all the cpp files to the rnworkletcore header dir in build which is not success.
> 
> Here's a patch which will copy the files individual, this fixed issue for me. FileLocation: "../node_modules/react-native-worklets-core/android/build.gradle"
> 
> <img alt="Screenshot 2024-09-03 at 12 08 19 PM" width="756" src="https://private-user-images.githubusercontent.com/14840130/363867616-a64fe165-8b3e-4dfe-a59d-b0273c481a96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTY3MTAsIm5iZiI6MTcyNjExNjQxMCwicGF0aCI6Ii8xNDg0MDEzMC8zNjM4Njc2MTYtYTY0ZmUxNjUtOGIzZS00ZGZlLWE1OWQtYjAyNzNjNDgxYTk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDA0NDY1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlODkwMTEyMDhkODJiNGZkMWZlOGMyODRlNDgzYmJiN2YwNWNhZjg0MGY3OGEyNzdmNDBmOWJiNzA1YjhmOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EI4O1XXZ-Q5zcnE-dVlVJNH1IXEj9NaL5LJ0hHm5PAg">

it worked for me.

react-native-worklets-core: 1.3.3

**Comment by dibakar-appinventiv** [comment]
> Not sure whethere is's an issue with gradle version, filetree filter method seems not working There's an prepare header task in the gradle file which copy all the cpp files to the rnworkletcore header dir in build which is not success.
> 
> Here's a patch which will copy the files individual, this fixed issue for me. FileLocation: "../node_modules/react-native-worklets-core/android/build.gradle"
> 
> <img alt="Screenshot 2024-09-03 at 12 08 19 PM" width="756" src="https://private-user-images.githubusercontent.com/14840130/363867616-a64fe165-8b3e-4dfe-a59d-b0273c481a96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0OTUxODEsIm5iZiI6MTcyOTQ5NDg4MSwicGF0aCI6Ii8xNDg0MDEzMC8zNjM4Njc2MTYtYTY0ZmUxNjUtOGIzZS00ZGZlLWE1OWQtYjAyNzNjNDgxYTk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDA3MTQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyOWJhZDE4NjA5NzhhNTVjNTQxYzM5MjY0YWM1OWRmNzhlNjMzZGIxYzExNmI3MmM0OWM0YjdhMmI2OWNiOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.G6jkbMSi9JqiENKfWCDULdvr1w0bi1S9MrqCMuth-m0">

How did you arrived at this solution ?

**Comment by sandeep36butte** [comment]
@dibakar-appinventiv Just checked for the build script function which is responsible for copying the cpp files to header folder in build directory.
Since on running the build with stack-trace  the error clearly states the files not present in the header folder inside the build directory.

**Comment by genesiscz** [comment]
@MuazzezA Why did you close this? Did you find a solution? I guess the issue should be still open if package did not solve the issue directly

**Comment by Rakedd** [comment]
Easier to close than to fix

**Comment by MuazzezA** [comment]
@genesiscz 
No, I couldn't solve it, but I no longer need this development. 
I suggest you change the versions and try.

@Rakedd _You shouldn't be prejudiced_, I have to deal with many problems and I closed it. Since I think it varies depending on the system, even if I solve it, it may not work for someone else.

**Comment by Rakedd** [comment]
@MuazzezA yesterday i spent all day trying to fix it and couldn't. At the end of the day i commented this out of anger . I apologize i should've been angry at myself because i did not manage to fix it.

**Comment by genesiscz** [comment]
This solved that for some reason 

https://github.com/expo/expo/issues/28309#issuecomment-2405560702
